Discovery and use of pylint config file for pylint lint plugin

jeeger opened the pull request:
I've added functionality to make the `pylint` plugin find the `pylintrc` config file and use it. I've added a `rcfile` plugin option for the pylint plugin that is populated with the path of any found pylint config file.

Also included is a test that checks whether the config file is passed correctly.

Settings for other plugins are broken out from their config file, but the number of settings for pylint is huge, and passing all options from the config file into the `flags` argument of `Pylint.lint()` would require a huge amount of work. This is a useful first implementation.

Commit:
Added pylint config parser to config.py

Commit:
Implented --rcfile option for Pylint linter.

Commit:
Setting rcfile option from available files in Pylint config.

Commit:
Added test for pylint with config file.

palantirtech commented:
Thanks for your interest in [palantir/python-language-server](https://github.com/palantir/python-language-server), @jeeger! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

Commit:
Linter fixes for CircleCI

Commit:
Added some debug output to pylint plugin.

Commit:
Fixed bug in Pylint config finding.

The config key used was the string 'settings.plugins.pylint', instead
of correct nested dictionaries.

Commit:
Changed pylint config search to search down from project root.

Normally, config files are searched for in ascending manner (i.e.
moving towards the root folder). However, in our configuration phase,
we don't know anything about any python files. Thus, root and document
path are identical, and this means that the config file can only be in
the project root directory.

The pylintrc file can be located in the current working
directory however, so we need to search *down* into subfolders.
Currently, we search at most 4 folders deep by default, but this can
be modified.

Commit:
Removed UTF-8 characters from log message.

Python2 can't handle unicode transparently.

Commit:
Linter fixes.

ccordoba12 commented:
Closing in favor of #539.
